src/parser.h

```diff
@@ -43,6 +43,7 @@ struct cmark_parser {
   /* Options set by the user, see the Options section in cmark.h */
   int options;
   bool last_buffer_ended_with_cr;
+  cmark_llist *syntax_extensions;
```

Comment:
I'd add a field `struct cmark_syntax_extension *next` directly to `struct cmark_syntax_extension` and change this field to point to the first extension. Then there's no need for `cmark_llist` at all.